db/write_batch_internal.h

```diff
@@ -174,7 +174,8 @@ class WriteBatchInternal {
       bool ignore_missing_column_families = false, uint64_t log_number = 0,
       DB* db = nullptr, bool concurrent_memtable_writes = false,
       SequenceNumber* next_seq = nullptr, bool* has_valid_writes = nullptr,
-      bool seq_per_batch = false, bool batch_per_txn = true);
+      bool seq_per_batch = false, bool batch_per_txn = true,
+      bool avoid_memtable_flush = false);
```

Comment:
I'm a little concerned about adding another parameter with default value. I know existing code has been using this, but I feel we should try to regard it as a last resort when we run out of other solutions.
How about add another function named `InsertIntoWithoutSchedulingFlush`? Then we can have a class named `MemTableInserterNoFlush` (maybe a different name) extending `MemTableInserter` and overriding the `CheckMemTableFull` method (need to make it virtual). Virtual function call has some performance overhead, but in this case should not be the bottleneck because the secondary instance should be bound by how often and how fast we read from the logs. Then `DBImplSecondary::RecoverLogFiles` can call `WriteBatchInternal::InsertIntoWithoutSchedulingFlush`.